--- replacer.plin.orig	2018-09-02 00:21:49.317932938 +0000
+++ replacer.plin	2018-09-14 23:32:58.121659133 +0000
@@ -33,11 +33,11 @@
 while (<>)
 {
     chop;
-    split;
+    @v = split(' ', $_);
 
 #    print STDERR "bad line: " . $_ . "\n" unless $#_ == 3;
 
-    $p = $_[$#_];
+    $p = $v[$#v];
 
 #    if ( $p =~ /n=\"[0-9]+:[0-9]+\"/ ) # temp fix!
 #    {
@@ -56,9 +56,9 @@
 
     $p = $M{'?'} unless $p; 
 
-    $_[$#_] = $p;
+    $v[$#v] = $p;
 
-    $_ = join (" ", @_);
+    $_ = join (" ", @v);
 
     print $_ . "\n";
 }
